ansible/roles/real-postfix/templates/postfix/virtual-alias-maps-ldap.cf.j2

```diff
@@ -0,0 +1,22 @@
+# Find any vhost with a name or alias matching the domain of the
+# e-mail address.  We're queried with an entire e-mail address, but
+# are only interested in checking whether the domain portion
+# corresponds to a vhost; we'll simply deliver any mail for the vhost
+# to its owner, appending the original lefthand side of the address as
+# an extension.  %d extracts only the domain.  We don't match the
+# scripts.mit.edu vhost here because we don't want to first resolve an
+# arbitrary address to a scripts account, and then end up sending
+# their mail to the owners of the scripts.mit.edu vhost.  The uid
+# attribute, generated by the CoS template
+# cn=vhostOwnerCoS,ou=VirtualHosts,dc=scripts,dc=mit,dc=edu, is the
+# name of the locker that owns the vhost.  Protocol version 3 is
+# necessary to use ldapi.
+
+server_host = {{ ldap_server }}
+search_base = ou=VirtualHosts,dc=scripts,dc=mit,dc=edu
+query_filter = (&(objectClass=scriptsVhost)(|(scriptsVhostName=%d)(scriptsVhostAlias=%d))(!(scriptsVhostName=scripts.mit.edu))(|{% for ip in ansible_all_ipv4_addresses %}(scriptsVhostPoolIPv4={{ip}}){% endfor %}))
```

Comment:
I fear my ongoing fondness for line breaks is once more flaring up.

Reply:
I'm also going to observe that I find it slightly disconcerting that Postfix uses %s and %d since I'm pretty used to printf, so %d "should" be an int, but this is obviously not your fault (and if I were more used to, say, `strftime` and less used to `printf` it'd probably feel less weird).